packages/contractkit/src/wallets/signers/azure-hsm-signer.ts

```diff
@@ -0,0 +1,50 @@
+import { ensureLeading0x, trimLeading0x } from '@celo/utils/lib/address'
+import * as ethUtil from 'ethereumjs-util'
+import { getHashFromEncoded, RLPEncodedTx } from '../../utils/signing-utils'
+import { AzureKeyVaultClient } from './azure-key-vault-client'
+import { Signer } from './signer'
+
+/**
+ * Signs the EVM transaction using an HSM key in Azure Key Vault
+ */
+export class AzureHSMSigner implements Signer {
+  private keyVaultClient: AzureKeyVaultClient
+  private keyName: string
+
+  constructor(keyVaultClient: AzureKeyVaultClient, keyName: string) {
+    this.keyVaultClient = keyVaultClient
+    this.keyName = keyName
+  }
+
+  async signTransaction(
+    addToV: number,
+    encodedTx: RLPEncodedTx
+  ): Promise<{ v: string; r: string; s: string }> {
```

Comment:
Any reason for `{ v: string; r: string; s: string }` in the "lower level" and `{ v: number; r: Buffer; s: Buffer }` higher up?

Reply:
It was originally based on what the caller needed but I like the idea of having a uniform interface and having the caller modify the value to suit their needs. {number, Buffer, Buffer} is a stronger interface so I'll go with that.